### ISSUE TITLE
If username contains special symbols GetUser() method fails

### ISSUE THREAD

**Issue by nivanov-stitan** [okta/okta-sdk-dotnet#533]
When I pass username with special symbols, for instance `#abcd`, `ab/cd`, ... 
`await oktaClient.Users.GetUserAsync()` is throw an exception.

**Comment by andriizhegurov-okta** [comment]
Hi @nivanov-stitan, what exception do you see exactly? 
Do you mean the `Username` profile field or is it another field with a similar name? The `Username` field should not accept special symbols, it only allows data in email format.

**Comment by nivanov-stitan** [comment]
Hi @andriizhegurov-okta 
>Do you mean the Username profile field or is it another field with a similar name? 
Yes, I mean Username which uses to login.
> The Username field should not accept special symbols, it only allows data in email format.
It's incorrect. It depends on User Profile settings in Okta org. We changed there that username can have non email format. And I can be able to create there users with special symbols.
It seems special symbols should be encoded before send HTTP request.

**Comment by andriizhegurov-okta** [comment]
@nivanov-stitan thanks for the clarification, I'll add an item to our backlog to investigate the issue.

**Comment by nivanov-stitan** [comment]
oktaClient.Users.GetUserAsync(".") returns error 
Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'System.Collections.Generic.IDictionary`2[System.String,System.Object]' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.
```
To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.
Path '', line 1, position 1.
```

**Comment by nivanov-stitan** [comment]
oktaClient.Users.GetUserAsync("ab/cd") returns error 
`The endpoint does not support the provided HTTP method (405, E0000022)`

**Comment by nivanov-stitan** [comment]
oktaClient.Users.GetUserAsync("lkj;lj") returns error
`Not found: Resource not found: lkj (User) (404, E0000007)`
By some reason it searches **lkj** instead of **lkj;lj**

**Comment by nivanov-stitan** [comment]
oktaClient.Users.GetUserAsync("Aph\DuncanC") returns error
`An error occurred deserializing the error body for response code 400. See the inner exception for details.`
The problem in backslash characters "\"

**Comment by nivanov-stitan** [comment]
Could you give some ETA for this critical bugs?
Thanks.

**Comment by andriizhegurov-okta** [comment]
@nivanov-stitan Thank you for your investigation and I apology for the delay. 
Unfortunately I can't provide an exact ETA for the bug, but I've prioritized it and moved it to the next sprint, so it should be taken into work soon.

**Comment by nivanov-stitan** [comment]
@andriizhegurov-okta Thank you.
We are using this Okta SDK in our application and the issues above doesn't allow us to manage users that contains such characters in username.

**Comment by andriizhegurov-okta** [comment]
@nivanov-stitan the issue is not specific to .Net SDK. It's reproducible from Postman using [Users API](https://developer.okta.com/docs/reference/api/users/) Postman collection.
[Get User with login](https://developer.okta.com/docs/reference/api/users/#get-user-with-login) API doesn't seem to work with some special characters. 

According to the [doc](https://developer.okta.com/docs/reference/api/users/#get-user-with-login): 

> When fetching a user by login, [URL encode (opens new window)](http://en.wikipedia.org/wiki/Percent-encoding)the request parameter to ensure special characters are escaped properly. Logins with a / character can only be fetched by id due to URL issues with escaping the / character.

So this is the expected behavior for slash symbols, but I can confirm that the other cases you've mentioned don't work either. 

As a workaround you can [get users by id](https://developer.okta.com/docs/reference/api/users/#get-user-with-id), not user names.
Please let me know if this unblocks you.

**Comment by nivanov-stitan** [comment]
@andriizhegurov-okta 
Hi.
We have some cases when we have not user id and need to check by username. For instance, in the initial migration we are checking that user with particular name exists or not exists in Okta org and do some logic. In that case we have no user id.

**Comment by andriizhegurov-okta** [comment]
@nivanov-stitan We're investigating the issue internally. 

Meanwhile you can try another workaround. 
To get user by login you can use `ListUsers` function with search parameter like so:

```csharp
var userLogin = "lkj;lj";
var users = await client.Users.ListUsers(search: $"profile.login eq \"{userLogin}\"").ToListAsync();
```

This works for semicolons (but not for slashes).

**Comment by laura-rodriguez** [comment]
Hi @nivanov-stitan,

I want to provide an update on this issue. 
We reported this issue to the internal team responsible for the users' API, and they will look into it soon. I'll provide more updates when I have more to share. Thanks for your patience!

**Comment by joserpkkr** [comment]
Hi, just wanted to note that I'm having this same issue when some users that have the '#' symbol on their email, so it would be nice to have a solution/workaround for this scenarios
